stubs/datasets/__init__.pyi

```diff
@@ -0,0 +1,7 @@
+import pandas as pd
+
+class Dataset:
+    @classmethod
+    def from_pandas(cls, df: pd.DataFrame) -> Dataset: ...
+
+class DatasetDict(dict): ...
```

Comment:
I have not used stubs before but seems we can add typing to other libs using stubs. What is this a stub for?

Reply:
It's probably not clear because HF's python package for their datasets is actually just `datasets`. This are stubs for the classes we use. I had failing lint checks from mypy without this.

Reply:
We haven't used stubs for all our other modules and seems non standard for this project. Can you just move the typing info into `datasets.py`

Reply:
LGTM - we resolve this offline. @goeffthomas will add comments to the file explaning that this stub is for the HF package called `datasets`.